src/Theme_Command.php

```diff
@@ -436,9 +437,19 @@ protected function get_item_list() {
 				$network_enabled = array();
 		}
 
+		$all_update_info = $this->get_update_info();
+		$checked_themes  = ( isset( $all_update_info->checked ) ) ? $all_update_info->checked : array();
+
+		if ( ! empty( $checked_themes ) ) {
+			foreach ( $checked_themes as $slug => $version ) {
+				$theme_version_info[$slug] = $this->is_theme_version_valid( $slug, $version );
+			}
+		}
+
 		foreach ( wp_get_themes() as $key => $theme ) {
 			$file = $theme->get_stylesheet_directory();
-			$update_info = $this->get_update_info( $theme->get_stylesheet() );
+
+			$update_info = ( isset( $all_update_info->response[$theme->get_stylesheet()]) && null !== $all_update_info->response[$theme->get_stylesheet()] ) ? (array) $all_update_info->response[$theme->get_stylesheet()] : null;
```

Comment:
WPCS - spaces around non-literal array indexes:
```suggestion
			$update_info = ( isset( $all_update_info->response[ $theme->get_stylesheet() ] ) && null !== $all_update_info->response[ $theme->get_stylesheet() ] ) ? (array) $all_update_info->response[ $theme->get_stylesheet() ] : null;
```